pkg/proctree/events_processing.go

```diff
@@ -0,0 +1,37 @@
+package proctree
+
+import (
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+// ProcessEvent update the process tree according to arriving event
+func (tree *ProcessTree) ProcessEvent(traceeEvent *trace.Event) error {
+	switch traceeEvent.EventName {
+	case "sched_process_fork":
+		return tree.processForkEvent(traceeEvent)
+	case "sched_process_exec":
+		return tree.processExecEvent(traceeEvent)
+	case "sched_process_exit":
+		return tree.processExitEvent(traceeEvent)
+	case "exit", "init_namespaces":
+		return nil
+	default:
+		return tree.processDefaultEvent(traceeEvent)
+	}
+}
+
+// processDefaultEvent tries to expand the process tree in case of lost events or missing general information
+func (tree *ProcessTree) processDefaultEvent(event *trace.Event) error {
+	process, err := tree.getProcess(event.HostProcessID)
+	if err != nil {
+		process = tree.addGeneralEventProcess(event)
+		process.addThread(event.HostThreadID)
+	} else if process.Status.Contains(uint32(hollowParent)) {
```

Comment:
Also, status is a transaction flag in this case, right ? You should use a lock/mutex acquire() before status flag check AND a release() after the operation you do (fillHollowParentProcess...) because of the flag existence.

Possibly the same in all other .Status.Contains situations...

If not, there might be a race in between the conditional logic execution and the flag status existence. I searched for that in the commit adding the RW map protection but couldn't find.